### ISSUE TITLE
ISO8601 datetime format has a variable length, we expect constant length

### ISSUE THREAD

**Issue by michwill** [nucypher/nucypher#906]
Throughout our code we use iso8601 datetieme format, made by `maya`. Maya returns it with microseconds, unless number of microseconds is 0. So, the result has a variable length:
```
In [2]: t = maya.now()                                                        

In [3]: t.iso8601()                                                           
Out[3]: '2019-04-03T22:15:05.400470Z'

In [4]: maya.MayaDT(t.epoch).iso8601()                                        
Out[4]: '2019-04-03T22:15:05Z'
```

If we pass the datetime in "0 microseconds" format to our API, it will result in Ursula failing to parse.
Example: replace `(maya.now() + datetime.timedelta(days=3)).iso8601()` with `maya.MayaDT((maya.now() + datetime.timedelta(days=3)).epoch).iso8601()` for expiration in `tests/characters/test_character_control.py`

The result:
```
  File "/home/michwill/Projects/nucypher-kms/nucypher/network/server.py", line 221, in consider_arrangement
    arrangement = Arrangement.from_bytes(request.data)
  File "/home/michwill/Projects/nucypher-kms/nucypher/policy/models.py", line 89, in from_bytes
    alice_pubkey_sig, arrangement_id, expiration_bytes = cls.splitter(arrangement_as_bytes)
  File "/home/michwill/.local/share/virtualenvs/nucypher-kms-5SSQ9ZHE/lib/python3.6/site-packages/bytestring_splitter/__init__.py", line 57, in __call__
    self.message_types, len(self), len(splittable)))
bytestring_splitter.BytestringSplittingError: Wrong number of bytes to constitute message types [(None, <class 'umbral.keys.UmbralPublicKey'>, 33, {}), (None, <class 'bytes'>, 32, {}), (None, <class 'bytes'>, 27, {})] - 
                    need 92, got 85 
 Did you mean to return the remainder?
```

**Comment by KPrasch** [comment]
Internally, let's handle grant expiration as epoch time, other than accepting ISO8601 from controllers and entry points. I'm encountering this now in PR #984 .

**Comment by tuxxy** [comment]
I'm very -1 on epoch time. ISO8601 datetimes are far better and way more flexible. They give us the option to potentially do far more than a plain epoch timestamp.

**Comment by cygnusv** [comment]
What we can do is to use a variable length bytestring splitter, instead. That way we can still use iso 8601.

**Comment by KPrasch** [comment]
Currently - internally we use `maya.DT` objects everywhere except for the following places we use ISO8601 Strings:
  - CLI expiration `nucypher alice grant --expiration` ( Alice Controller )
  - `WebController` Request ( Alice Controller )
  - Arrangement Serialization `Arrangement.__bytes__`  
  - Moe Monitoring Websocket Push of Fleet State
(I think there is also some outstanding work to do regarding timestamping interfaces.. #589 ..?)

I do not want to excise the use of ISO8601 strings altogether (in fact they are very well suited for CLI and `WebController` requests), but I do want to reduce some of the complexity caused by handling datetime objects internally, and fix the original Issue here which is caused by `Arrangement.__bytes__`.  ISO8601 strings produced by `maya` or python `datetime` are variable length, resulting in (sometimes) failed serialization.

At the core of nucypher policies - on-chain periods are epoch-based in the `PolicyManager` contract, so we end up needing to do things like: <https://github.com/nucypher/nucypher/blob/0c8c7fb87a8c73657bbc5aed09462f4ae04ad60d/nucypher/blockchain/eth/utils.py#L25> when creating policies, calculating stakes, etc. 

- What advantages of ISO8601 strings do we want to preserve?
- If we continue to use ISO8601 strings for `Policy` + `Arrrangement` serialization, How do we handle the variable length issue?

**Comment by michwill** [comment]
I am ok with either epoch or ISO8601. However if we choose to use ISO, I propose to round to even number of seconds (so that we have constant length of the string). Revoke milliseconds probably don't matter, and maya will return `2019-04-03T22:15:05Z` instead of `2019-04-03T22:15:05.000000Z` with 1/million probability (so the length of the string _will_ change)

**Comment by cygnusv** [comment]
I'm trying to fix #1134, but I've stumbled upon this again and I think a really easy solution is simply to use a `VariableLengthBytestring` to serialize the ISO8601 dates and call it a day. I'm on it.

**Comment by michwill** [comment]
`VariableBytestringSplitter` is good but... Really, should we add those microseconds at all? I'd rather chop them off when serializing. And tbh I don't even remember seeing microseconds in the ISO8601 spec itself
